db/managed_iterator.cc

```diff
@@ -93,6 +93,7 @@ ManagedIterator::~ManagedIterator() {
     snapshot_created_ = false;
     read_options_.snapshot = nullptr;
   }
+  UnLock();
```

Comment:
Debug diagnostics was (correctly) complaining that the mutex is being destroyed while owned because it is acquired at the start of the __dtor. The proper fix would be to remove the Lock() line since the object must not be ordinarily a subject of MT access at the time of destruction but we apparently decided to play along in this case and simply unlocked it.